### DISCUSSION TITLE
A photo editor with no crop tool.  Is this a joke?

### DISCUSSION THREAD

**Discussion by StoicVoyager** [tannerhelland/PhotoDemon#578] [General]
Title says it all.

**Comment by manfromarce** [comment]
You need to select an area and then click Image > Crop selection. Paint.net works the same way

**Reply by StoicVoyager** [reply]
So there is indeed no crop tool.  Ok.

**Comment by tannerhelland** [comment]
Hello @StoicVoyager .  I am the developer of PhotoDemon.  Because you have no code repositories of your own, I will assume you are new to GitHub and extend you some benefit of the doubt.

Before posting a new question, it can be useful to search a GitHub repository first.  A search here for "Crop" which would have immediately turned up this issue from earlier this year:

[Dedicated Crop tool](https://github.com/tannerhelland/PhotoDemon/issues/530)

As discussed in that issue, I hope to add a dedicated `Crop tool` soon.  A `Crop tool` is just a convenience wrapper around the rectangular selection tool, which you have already been directed toward.  PhotoDemon currently skews toward advanced users and some settings may not be immediately obvious to beginners.  (Photoshop had a similar problem in the beginning.  It didn't introduce a `Crop tool` until version 6, I believe.)

I apologize that it's taken me some time to add this feature.

In the future, I think *all* developers of open-source software would appreciate kindness and patience regarding the software we make available for free.

**Reply by finallyfree0126** [reply]
May I suggest adding this to the toolbar so it is a one-button solution if it is possible?  I have to imagine it is one of, if not the most common edit done.  It may seem small to make it two clicks but I have to crop sometimes a thousand photos a day so an extra button adds up in time and stress on the wrist and hand.  I am looking for a replacement for Paint.Net but that prevents me from making the switch.  i like the tool otherwise.  Good luck on future development!

**Reply by hi5** [reply]
Dedicated "Crop tool" or not, it will always be two stage operation: selection and cropping. 

All that is missing it a button on the toolbar, but there is a shortcut in case you haven't seen it yet. 
So open the image, press <kbd>s</kbd> for select if not already on the select tool, then select (obviously), then press the shortcut, your hand can be on the keyboard <kbd>ctrl</kbd>+<kbd>shift</kbd>+<kbd>x</kbd> - if that combo is too uncomfortable, you can use a tool like [AutoHotkey](https://www.autohotkey.com/) to simply remap it to something [you like better](https://www.autohotkey.com/docs/v2/Hotkeys.htm#Context).

Depending on your cropping "needs" think about ways to automate it, either using a macro in PD or external programs like image magick (or AutoHotkey) which you can script.

**Reply by Edinho367** [reply]
We users often take for granted the effort and dedication that goes into creating software that actually makes our work more efficient. 

Most people rarely take the time to appreciate, let alone encourage, developers like you who dedicate their free time to making something valuable available to others. 

Without free software, technology might not be as accessible to everyone today. The alternative – a world dominated by a few powerful corporations – would be quite daunting.

Thank you for your hard work and for staying patient with us users, despite the occasional criticism and negative feedback; like the bitter question where this thread started.... We appreciate your commitment to improving PD and making a valuable contribution to the tech community.

Cannot wait for the next PD version and the improvements!


Sorry for my english, i'm not native...

**Comment by tannerhelland** [comment]
The latest nightly builds now included a dedicated crop tool.  Thank you to everyone for their patience while I worked on this.

The new crop tool supports a lot of helpful features, including (optional) built-in guides, size and aspect ratio presets, non-destructive cropping, "allow enlarging", interactive resizing, image vs layer targeting and much more.

![Image](https://github.com/user-attachments/assets/e4bb4545-dd2e-441c-b288-952b53c9e859)

You are welcome to test it out in the latest nightly builds, or you can wait for the next stable release (currently targeted for next month, e.g. v2025.4).

Thanks!

**Reply by Edinho367** [reply]
Veeery nice!!! 😎🤩😆

Thanks a lot!!! 👍👊

**Reply by hi5** [reply]
Nicely executed... 🎨
